packages/@aws-cdk/pipelines/test/compliance/basic-behavior.test.ts

```diff
@@ -25,34 +25,44 @@ behavior('stack templates in nested assemblies are correctly addressed', (suite)
     const pipeline = new LegacyTestGitHubNpmPipeline(pipelineStack, 'Cdk');
     pipeline.addApplicationStage(new OneStackApp(app, 'App'));
 
-    THEN_codePipelineExpectation();
+    Template.fromStack(pipelineStack).hasResourceProperties('AWS::CodePipeline::Pipeline', {
```

Comment:
The point here was that we would have the same assertions for both types of pipelines, so that we know they produce equivalent outputs.

Duplicating and twiddling the assertions does the opposite.

I see you're doing this to avoid having to implement `stringLike()`. Convince me that's worth the trade-off?

Reply:
Why would we change the legacy pipeline API anymore? As far as I can tell, the legacy tests and assertions are here forever (until deleted).

Reply:
> so that we know they produce equivalent outputs

Also, why is this the goal of the test? They may produce different outputs as the (new) API evolves.

Reply:
> They may produce different outputs as the (new) API evolves.

Fair, and then the new tests we would add for this wouldn't be compliance tests or they would be written to explicitly indicate that a new feature is not available in the legacy API.

> Why would we change the legacy pipeline API behaviour anymore?

We could change things that don't have anything to do with the outward behavior, such as hashes or internal identifiers. I'd prefer not to have to update test snapshots for things that aren't actually the point of the test.

-----

Not sure why you're refusing to copy/paste a relatively simple matcher over, but I guess we won't see eye to eye on this. 

I won't block you from merging the new features into `assertions`. I will approve if you take out the changes to the pipelines tests.

Reply:
Another strategy I could live with is implementing `stringLike()` in the pipelines module as custom matcher.

Reply:
> we would add for this wouldn't be compliance tests

Ah, I failed to notice the folder named 'compliance'